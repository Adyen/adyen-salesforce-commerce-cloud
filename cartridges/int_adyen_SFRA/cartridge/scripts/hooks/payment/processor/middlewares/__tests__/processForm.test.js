"use strict";

var _processForm = require("mockData/processForm");

/* eslint-disable global-require */
var processForm;
var req;
beforeEach(function () {
  processForm = require('../processForm');
  jest.clearAllMocks();
  req = {
    form: {
      adyenPaymentMethod: 'mockedPaymentMethod',
      adyenIssuerName: 'mocked_issuer_name',
      brandCode: 'scheme'
    },
    currentCustomer: {
      raw: {}
    }
  };
});
describe('processForm', function () {
  it('should return error when credit card validation fails', function () {
    var processFormOutput = processForm(req, (0, _processForm.getPaymentForm)(), {});
    expect(processFormOutput).toMatchSnapshot();
  });
  it('should return viewData', function () {
    req.form.storedPaymentUUID = 'mockedUUID';
    var processFormOutput = processForm(req, (0, _processForm.getPaymentForm)(), {});
    expect(processFormOutput).toMatchSnapshot();
  });
  it('should return viewData when authenticated and registered', function () {
    var paymentForm = (0, _processForm.getPaymentForm)();
    var uuid = 'mocked_id';
    var paymentInstrument = (0, _processForm.getPaymentInstrument)(uuid);
    paymentForm.creditCardFields.selectedCardID = {
      value: uuid
    };
    req.form.brandCode = 'not_scheme';
    req.form.securityCode = 'mocked_security_code';
    req.currentCustomer.wallet = {
      paymentInstruments: [paymentInstrument]
    };
    req.currentCustomer.raw = {
      authenticated: true,
      registered: true
    };
    var processFormResult = processForm(req, paymentForm, {});
    expect(processFormResult).toMatchSnapshot();
  });
});