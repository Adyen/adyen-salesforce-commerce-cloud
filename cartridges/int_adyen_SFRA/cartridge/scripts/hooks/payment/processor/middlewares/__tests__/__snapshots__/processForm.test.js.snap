// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processForm should return error when credit card validation fails 1`] = `
Object {
  "error": true,
  "fieldErrors": Object {
    "creditCardErrors": "mockedCreditCardErrors",
  },
}
`;

exports[`processForm should return viewData 1`] = `
Object {
  "error": false,
  "viewData": Object {
    "paymentInformation": Object {
      "adyenIssuerName": "mocked_issuer_name",
      "adyenPaymentMethod": "mockedPaymentMethod",
      "cardNumber": "mockedCardNumber",
      "cardType": "mockedCardType",
      "isCreditCard": true,
      "stateData": "mockedStateData",
    },
    "paymentMethod": Object {
      "htmlName": "mockedPaymentMethod",
      "value": "mockedPaymentMethod",
    },
    "saveCard": true,
  },
}
`;

exports[`processForm should return viewData when authenticated and registered 1`] = `
Object {
  "error": false,
  "viewData": Object {
    "0": "m",
    "1": "o",
    "2": "c",
    "3": "k",
    "4": "e",
    "5": "d",
    "6": "_",
    "7": "i",
    "8": "d",
    "paymentInformation": Object {
      "0": "m",
      "1": "o",
      "2": "c",
      "3": "k",
      "4": "e",
      "5": "d",
      "6": "_",
      "7": "i",
      "8": "d",
      "adyenIssuerName": "mocked_issuer_name",
      "adyenPaymentMethod": "mockedPaymentMethod",
      "cardNumber": "mocked_cc_number",
      "cardType": "mocked_cc_type",
      "creditCardToken": "mocked_raw_cc_token ",
      "expirationMonth": "mocked_cc_exp_month",
      "expirationYear": "mocked_cc_exp_year",
      "isCreditCard": false,
      "securityCode": "mocked_security_code",
      "stateData": "mockedStateData",
    },
    "paymentMethod": Object {
      "htmlName": "mockedPaymentMethod",
      "value": "mockedPaymentMethod",
    },
    "saveCard": true,
  },
}
`;
